app/controllers/questions_controller.rb

```diff
@@ -34,10 +34,22 @@ def unanswered
     render 'index'
   end
 
+  def destroy
+    question = Question.find_by(id: params[:id])
+
+    if current_user == question.user
+      question.destroy
+      flash[:notice] = "You've successfully deleted #{question.title}"
+      redirect_to root_path
+    else
+      flash[:error] = "Delete failed. You are not the owner of this question."
+      redirect_to question
+    end
+  end
+
```

Comment:
This if/else should not be necessary.  The accepted way to manage authorizations is using before_action methods to bar access to the action (destroy, in this case) completely.  You shouldn't have to manually check user_ids in the action.

See the link I sent you for an example of requiring a logged in user.  The same method could be used to check the current_user.id against the record's owner ID and simply redirect requests away from the protected route.